### PR TITLE
[test] Remove panic from collections unit tests

### PR DESCRIPTION
**Pull request by iyzhang**
This PR addresses issue #207 for all of the unit tests in src/rust/collections. It also adds a return value to every test and replaces assert with ensure and assert_eq with the new ensure_eq macro.

### CODE REVIEW

File: src/rust/collections/intrusive/intrusive_queue.rs [hunk 1]
```diff
@@ -170,13 +170,13 @@ mod tests {
...
     #[test]
-    fn fifo_order() {
+    fn fifo_order() -> Result<()> {
         // Create the queue.
         let mut test_iq: IntrusiveQueue<TestThingy> = IntrusiveQueue::new();
 
         // Verify: The queue should be empty.
-        assert!(test_iq.is_empty());
-        assert_eq!(test_iq.len(), 0);
+        anyhow::ensure!(test_iq.is_empty());
```
Comment (ppenna) [inline src/rust/collections/intrusive/intrusive_queue.rs:178]: Couldn't we just use `crate::ensure_eq!()`?

File: src/rust/collections/intrusive/intrusive_queue.rs [hunk 2]
```diff
@@ -191,8 +191,8 @@ mod tests {
         test_iq.push_element(element4);
 
         // Verify: The queue should now contain 4 elements.
-        assert_eq!(test_iq.is_empty(), false);
-        assert_eq!(test_iq.len(), 4);
+        anyhow::ensure!(!test_iq.is_empty());
```
Comment (ppenna) [inline src/rust/collections/intrusive/intrusive_queue.rs:194]: Couldn't we just use `crate::ensure_eq!()`?

File: src/rust/collections/intrusive/intrusive_queue.rs [hunk 3]
```diff
@@ -201,63 +201,73 @@ mod tests {
...
 
             // Verify length of queue is correct.
-            assert_eq!(test_iq.len(), 4 - check_data as usize);
+            crate::ensure_eq!(test_iq.len(), 4 - check_data as usize);
         }
 
         // Verify: The queue should be empty.
-        assert!(test_iq.is_empty());
-        assert_eq!(test_iq.len(), 0);
+        anyhow::ensure!(test_iq.is_empty());
```
Comment (ppenna) [inline src/rust/collections/intrusive/intrusive_queue.rs:214]: Couldn't we just use `crate::ensure_eq!()`?

File: src/rust/collections/raw_array.rs [hunk 4]
```diff
@@ -6,6 +6,7 @@
 //======================================================================================================================
 
 use crate::runtime::fail::Fail;
+use ::anyhow::Result;
```
Comment (ppenna) [inline src/rust/collections/raw_array.rs:9]: Why add this to mainline code?

File: src/rust/collections/ring.rs [hunk 5]
```diff
@@ -11,6 +11,7 @@ use crate::{
     collections::raw_array,
     runtime::fail::Fail,
 };
+use ::anyhow::Result;
```
Comment (ppenna) [inline src/rust/collections/ring.rs:14]: Why add this to mainline code?

**Review by ppenna — APPROVED** [review]
Looks good to me. I left some minor comments.

Thanks for working on this.

File: src/rust/collections/raw_array.rs (hunk 4 above)
Comment (iyzhang) [reply src/rust/collections/raw_array.rs:9]: I'll remove this

File: src/rust/collections/ring.rs (hunk 5 above)
Comment (iyzhang) [reply src/rust/collections/ring.rs:14]: also removed

**Pull request merged**